### PR TITLE
feat(flatten-library): New task to flatten library in build result

### PR DESCRIPTION
**Pull request by matz3**

### CODE REVIEW

File: packages/ui5-task-flatten-library/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+- For `src` files it removes the `/resources/${projectNamespace}/` part of the path (e.g. `dist/resources/my/lib/library.js` => `dist/library.js`)
+- All `test` files (`/test-resources/`) will be omitted from the build result
+- A warning is logged for `src` files that are not part of the project namespace
+
+## License
+
+This work is [dual-licensed](../../LICENSE) under Apache 2.0 and the Derived Beer-ware License. The official license will be Apache 2.0 but finally, you can choose between one of them if you use this work.
+
+When you like this stuff, buy [@vobu](https://twitter.com/vobu) a beer or buy [@pmuessig](https://twitter.com/pmuessig) or [@TheVivekGowda](https://twitter.com/TheVivekGowda) a coke when you see them.
```
Comment (vobu) [inline packages/ui5-task-flatten-library/README.md:62]: might wanna attribute some beerz to you too

File: packages/ui5-task-flatten-library/lib/flatten-library.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ */
+module.exports = async function ({ workspace, taskUtil, options }) {
+
+  if (!taskUtil.isRootProject()) {
+    // Flattening is only done when library is built as root project
+    log.info(`Skipping execution. Current project '${options.projectName}' is not the root project.`);
+    return;
+  }
+
+  if (!options.projectNamespace) {
```
Comment (vobu) [inline packages/ui5-task-flatten-library/lib/flatten-library.js:23]: please put the documentation for this config-time param into the `README` and/or explain its' usage in the extesion

**Review by vobu — CHANGES_REQUESTED** [review]
only minor stuff.
and just a hint: is the usage of task util in `@ui5/builder.tasks.TaskUtil` already documented? now that i've seen it being used, might be a really helpful tool for others for extension dev.
same goes for `DuplexCollection.write(file)`

File: packages/ui5-task-flatten-library/README.md (hunk 1 above)
Comment (matz3) [reply packages/ui5-task-flatten-library/README.md:62]: Haha, maybe 😉

File: packages/ui5-task-flatten-library/lib/flatten-library.js (hunk 2 above)
Comment (matz3) [reply packages/ui5-task-flatten-library/lib/flatten-library.js:23]: That's a default parameter that every task receives: https://sap.github.io/ui5-tooling/pages/extensibility/CustomTasks/#task-implementation

**Comment by matz3** [comment]
@vobu see https://sap.github.io/ui5-tooling/pages/extensibility/CustomTasks/#helper-class-taskutil and https://sap.github.io/ui5-tooling/api/module-@ui5_fs.DuplexCollection.html#write.

**Pull request merged**